projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts

```diff
@@ -0,0 +1,138 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  EventEmitter,
+  Input,
+  OnInit,
+  Output,
+  TemplateRef,
+  ViewEncapsulation
+} from '@angular/core';
+
+@Component({
+  selector: 'kal-list',
+  templateUrl: './kal-list.component.html',
+  styleUrls: ['./kal-list.component.sass'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class KalListComponent<T> implements OnInit {
+
+  /**
+   * Results list
+   */
+  results: T[];
+
+  /**
+   * Row templates
+   */
+  @Input() rowTemplate: TemplateRef<any>;
+
+  /**
+   * Datasource to give items list to the component
+   */
+  @Input() datasource: any;
+
+  /**
+   * Triggered when selection has changed
+   */
+  @Output() selectionChange = new EventEmitter<T>();
+  /**
+   * The selected item
+   */
+  private selectedItem = null;
+
+  /**
+   * Initials config
+   */
+  private initialsConfig: (item: T) => string = null;
+
+  /**
+   * Is the row disabled
+   */
+  private isDisabled: (item: T) => boolean = (item: T) => false;
+
+  constructor(private cdr: ChangeDetectorRef) {
+  }
+
+  /**
+   * Display initials in listing
+   */
+  @Input()
+  get initials(): (item: T) => string {
+    return this.initialsConfig;
+  }
+
+  set initials(value: (item: T) => string) {
```

Comment:
discriminant of items could be other than initials, should rename function name to reflect this possibility